src/fiat_shamir/constraints.rs

```diff
@@ -0,0 +1,448 @@
+use crate::fiat_shamir::{AlgebraicSponge, FiatShamirAlgebraicSpongeRng, FiatShamirRng};
+use crate::{overhead, Vec};
+use ark_ff::{BigInteger, PrimeField};
+use ark_nonnative_field::params::get_params;
+use ark_nonnative_field::{AllocatedNonNativeFieldVar, NonNativeFieldVar};
+use ark_r1cs_std::{
+    alloc::AllocVar,
+    bits::{uint8::UInt8, ToBitsGadget},
+    boolean::Boolean,
+    fields::fp::AllocatedFp,
+    fields::fp::FpVar,
+    R1CSVar,
+};
+use ark_relations::lc;
+use ark_relations::r1cs::{ConstraintSystemRef, LinearCombination, SynthesisError};
+use core::marker::PhantomData;
+
+/// Vars for a RNG for Fiat-Shamir purposes
+pub trait FiatShamirRngVar<F: PrimeField, CF: PrimeField, PFS: FiatShamirRng<F, CF>>:
```

Comment:
Do we plan to have other kinds of `FiatShamirRngVar` than `FiatShamirAlgebraicSpongeRngVar`?

Reply:
Eventually this will become a `SpongeGadget`, corresponding to the API in the `ark-sponge` crate